### PR TITLE
fixes express consoles not working right

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
fix: express console sends rockets
/:cl:

this code made it run the father console add interaction which made it not send a rocket but just add a request, it is now removed making it work again

### CODE REVIEW

**Comment by willox** [comment]
this allows ghosts use an unlocked express supply console if they somehow spoof tgui acts (which they can)

**Comment by Pugie** [comment]
Other stuff like mech orbital pads are broken and don't send either, could it be related?

**Review by Rohesie — CHANGES_REQUESTED** [review]
Don't remove the parent call, rework it at the base to support the behavior you want, instead.

File: code/modules/cargo/console.dm [hunk 1]
```diff
@@ -168,6 +168,8 @@
 				log_game("[key_name(usr)] accepted a shuttle loan event.")
 				. = TRUE
 		if("add")
+			if(istype(src, /obj/machinery/computer/cargo/express)) //prevents running this in the wrong way
+				return . = FALSE
```
Comment (Rohesie) [inline code/modules/cargo/console.dm:172]: `istype(src)` is almost never the right way to do things, what you want is a redefined proc
I assume you wanted to `return FALSE` so the rest of the proc doesn't happen.
The correct way h ere would be to move the things inside the `if("add)` check into a proc such as `/obj/machinery/computer/cargo/proc/add_id(id)`, which you redefine for `/obj/machinery/computer/cargo/express`

File: code/modules/cargo/console.dm [hunk 2]
```diff
@@ -17,7 +17,7 @@
 	/// var that tracks message cooldown
 	var/message_cooldown
 	var/list/loaded_coupons
-
+	var/express = FALSE
```
Comment (Rohesie) [inline code/modules/cargo/console.dm:20]: Could you document this variable?
Maybe give it a better name as well?
What does `express` mean?

File: code/modules/cargo/console.dm [hunk 3]
```diff
@@ -168,6 +168,8 @@
 				log_game("[key_name(usr)] accepted a shuttle loan event.")
 				. = TRUE
 		if("add")
+			if(express) //prevents running this in the wrong way
+				return . = FALSE
```
Comment (Rohesie) [inline code/modules/cargo/console.dm:172]: You want either `return FALSE` or `. = FALSE`, depending on if you want it to return here or not.

**Pull request closed without merging**